src/build/analyzer.ts

```diff
@@ -49,7 +49,7 @@ export class StreamAnalyzer extends Transform {
       this.allFragments.push(shell);
     }
 
-    if (entrypoint) {
+    if (entrypoint && !shell && fragments.length === 0) {
```

Comment:
The entrypoint can possibly import other things, but 1) they shouldn't and 2) we won't analyze or bundle it if there's a shell. This is part of the app structure we're defining in the docs. We can lint for this.